Generated react application with `--pascalCaseFiles` flag fails jest on case sensitive file systems

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
<!-- What is the behavior that currently you experience? -->
When generating a react application with `--pascalCaseFiles` the jest tests failed only in a CI environment that is case sensitive (i.e. Linux). Locally on mac, it worked fine because mac is case insensitive. 

CI (Jenkins with ubuntu pod) got the error 
Locally they passed. (Mac)

I am opening the issue to document since I have now hit this on more than one occasion and it took me too long to find the issue, since local could not reproduce. I will open a PR to resolve, it should be an easy fix.

## Expected Behavior
<!-- What is the behavior that you expect to happen? -->
<!-- Is this a regression? .i.e Did this used to be the behavior at one point?  -->
When using the `--pascalCaseFiles` flag the generated spec should import the App with `import App from './App';` to prevent this issue.

## Steps to Reproduce
<!-- Help us help you by making it easy for us to reproduce your issue! -->

<!-- Can you reproduce this on https://github.com/nrwl/nx-examples? -->
<!-- If so, open a PR with your changes and link it below. -->
<!-- If not, please provide a minimal Github repo -->
<!-- At the very least, provide as much detail as possible to help us reproduce the issue -->
https://github.com/nrwl/nx-examples/pull/195 has the example issue and steps to reproduce using a Docker container. I believe this is not reproducible on mac or with a volume mount on mac but is reproducible with all linux files systems.

### Failure Logs
<!-- Please include any relevant log snippets or files here. -->
`Cannot find module './app' from 'src/app/App.spec.tsx'`

```
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from './app';
    
at Resolver.resolveModule (../../node_modules/jest-runtime/node_modules/jest-resolve/build/resolver.js:324:11)
at Object.<anonymous> (src/app/App.spec.tsx:3:1)
```

### Environment
<!-- It's important for us to know the context in which you experience this behavior! -->
<!-- Please paste the result of `nx report` below! -->
```bash
Node : 16.14.0
OS   : linux x64
yarn : 1.22.17

nx : 13.8.1
@nrwl/angular : 13.8.1
@nrwl/cli : 13.8.1
@nrwl/cypress : 13.8.1
@nrwl/detox : undefined
@nrwl/devkit : 13.8.1
@nrwl/eslint-plugin-nx : 13.8.1
@nrwl/express : undefined
@nrwl/jest : 13.8.1
@nrwl/js : 13.8.1
@nrwl/linter : 13.8.1
@nrwl/nest : undefined
@nrwl/next : undefined
@nrwl/node : undefined
@nrwl/nx-cloud : 13.1.4
@nrwl/react : 13.8.1
@nrwl/react-native : undefined
@nrwl/schematics : undefined
@nrwl/storybook : 13.8.1
@nrwl/tao : 13.8.1
@nrwl/web : 13.8.1
@nrwl/workspace : 13.8.1
typescript : 4.5.5
rxjs : 6.5.5
---------------------------------------
Community plugins:
 @angular/animations: 13.2.2
 @angular/common: 13.2.2
 @angular/compiler: 13.2.2
 @angular/core: 13.2.2
 @angular/forms: 13.2.2
 @angular/platform-browser: 13.2.2
 @angular/platform-browser-dynamic: 13.2.2
 @angular/router: 13.2.2
 @ngrx/component-store: 13.0.1
 @ngrx/effects: 13.0.1
 @ngrx/entity: 13.0.1
 @ngrx/router-store: 13.0.1
 @ngrx/store: 13.0.1
 @angular-devkit/build-angular: 13.2.3
 @angular/cli: 13.2.3
 @angular/compiler-cli: 13.2.2
 @angular/language-service: 13.2.2
 @ngrx/store-devtools: 13.0.1
  ```

--- a/packages/react/src/generators/application/application.spec.ts
+++ b/packages/react/src/generators/application/application.spec.ts
@@ -399,6 +399,16 @@ Object {
       expect(appTree.exists('apps/my-app/src/app/App.spec.tsx')).toBeTruthy();
       expect(appTree.exists('apps/my-app/src/app/App.module.css')).toBeTruthy();
     });
+
+    it(`should use the correct case for file import in the spec file`, async () => {
+      await applicationGenerator(appTree, { ...schema, pascalCaseFiles: true });
+
+      const appSpecContent = appTree
+        .read('apps/my-app/src/app/App.spec.tsx')
+        .toString();
+
+      expect(appSpecContent).toMatch(/import App from '.\/App'/);
+    });
   });
 
   it('should generate functional components by default', async () => {
@@ -407,6 +417,16 @@ Object {
     const appContent = appTree.read('apps/my-app/src/app/app.tsx').toString();
 
     expect(appContent).not.toMatch(/extends Component/);
+  });
+
+  it(`should use the correct case for file import in the spec file`, async () => {
+    await applicationGenerator(appTree, { ...schema });
+
+    const appSpecContent = appTree
+      .read('apps/my-app/src/app/app.spec.tsx')
+      .toString();
+
+    expect(appSpecContent).toMatch(/import App from '.\/app'/);
   });
 
   it('should add .eslintrc.json and dependencies', async () => {

--- a/packages/react/src/generators/application/files/common/src/app/__fileName__.spec.tsx__tmpl__
+++ b/packages/react/src/generators/application/files/common/src/app/__fileName__.spec.tsx__tmpl__
@@ -3,7 +3,7 @@ import { render } from '@testing-library/react';
 import { BrowserRouter } from 'react-router-dom';
 <% } %>
 
-import App from './app';
+import App from './<%= fileName %>';
 
 describe('App', () => {
   it('should render successfully', () => {

